File: testdata/bench/rust/BenchmarkTest00085.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: XSS via script injection
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("data").unwrap_or(&String::new()).clone();
    let html = format!("<script>var data = '{}';</script>", param);
    HttpResponse::Ok().content_type("text/html").body(html)
}

File: testdata/bench/cpp/BenchmarkTest00085.cpp
#include <string>
#include <iostream>
#include <cstdlib>

int main() {
    char *input = getenv("QUERY_STRING");
    std::string html = "<html><body><p>" + std::string(input ? input : "") + "</p></body></html>";
    std::cout << "Content-Type: text/html\r\n\r\n" << html;
    return 0;
}
